capstone/client/src/components/StoresPage.js

```diff
@@ -70,6 +81,96 @@ class StorePage extends Component {
     });
   }
 
+  /* 
+   * Displays a dialog prompting the user to specify a name for the list that 
+   * is going to be saved. 
+   */
+  onShare = () => {
+    this.setState({
+      shareDialog: {
+        display: true,
+        shareButtonDisabled: true,
+        error: true,
+        errorText: "This is a required field."
+      },
+      errorMessage: null,
+    });
+  }
+
+  /*
+   * Checks to see if the "Email" field is empty. If the field is empty, an 
+   * error message is displayed and the save button is disabled. 
+   */
+  onTextFieldChange = (event) => {
+    if (event.target.value.trim() === '') {
+      this.setState({
+        shareDialog: {
+          display: true,
+          error: true,
+          errorText: "This is a required field.",
+          shareButtonDisabled : true,
+        },
+      });
+    } else {
+      this.setState({
+        email: event.target.value,
+        shareDialog: {
+          display: true,
+          shareButtonDisabled: false,
+        },
+      });
+    }
+  }
+
+  /* 
+   * Obtains the selected items from the checkbox list and makes a POST request to 
+   * /api/v1/share-via-email to save the specified list.
+   */
+  handleDialogSubmit = () => {
+    this.setState({
+      shareDialog: {
+        display: false,
+      },
+    });
+    axios.post(
+      '/api/v1/share-via-email',
+      { 
+        email : this.state.email,
+        html : "<h1>Test</h1>",
```

Comment:
Also put a subject? Can be a fake one for this PR.

Reply:
The front end shouldn't be constructing the HTML string. So we probably wanna change the content of this field anyway (to the list of stores or something)

Reply:
Yeah I want to alter it so that it takes a list of stores, for now though just to get it working I am passing HTML.